sqlglot/optimizer/rewrite_subqueries.py

```diff
@@ -84,3 +85,27 @@ def decorrelate_subqueries(expression):
             )
 
     return expression
+
+
+def expand_multi_table_selects(expression):
+    for from_ in expression.find_all(exp.From):
+        parent = from_.parent
+        where = parent.args.get("where")
+        for query in from_.args["expressions"][1:]:
+            alias = query.alias_or_name
+
+            conditions = [
+                condition
+                for condition in [
+                    column.find_ancestor(exp.EQ)
+                    for column in (where.find_all(exp.Column) if where else [])
+                    if column.text("table") == alias
+                ]
+                if condition
+            ]
+
+            for condition in conditions:
+                condition.replace(exp.TRUE)
+            parent.join(query, on=conditions, copy=False)
```

Comment:
I'm not that familiar with comma joins, and I know this is starting small, but will we need something like this?:

```python
if conditions:
    parent.join(query, on=conditions, copy=False)
else:
    parent.join(query, join_type="cross", copy=False)
```

Reply:
a join with conditions is a cross join